ush/regrid_gsiSfcIncr_to_tile.sh

```diff
@@ -0,0 +1,121 @@
+
+# CSD: temporary on hera, until scrip files added to fix dir. 
+TMP_FIX_FILES=/scratch2/BMC/gsienkf/Clara.Draper/gw_new_fix_files/
+
+export PGMOUT=${PGMOUT:-${pgmout:-'&1'}}
+export PGMERR=${PGMERR:-${pgmerr:-'&2'}}
+export REDOUT=${REDOUT:-'1>'}
+export REDERR=${REDERR:-'2>'}
+
+REGRID_EXEC=${HOMEgfs}/sorc/gdas.cd/build/bin/regridStates.x
+
+export PGM=${REGRID_EXEC}
+export pgm=${PGM}
+
+NMEM_REGRID=${NMEM_REGRID:-1}
+CASE_IN=${CASE_IN:-${CASE_ENS}}
+LFHR=${LFHR:-6}
+
+# get resolutions
+CRES_IN="${CASE_IN:1}"
+LONB_CASE_IN=$((4*CRES_IN))
+LATB_CASE_IN=$((2*CRES_IN))
+
+CRES_OUT="${CASE_OUT:1}"
+ntiles=6
+
+APREFIX_ENS="enkfgdas.t${cyc}z."
+
+LSOIL_INCR=${LSOIL_INCR:-2}
+
+n_vars=$(( LSOIL_INCR*2 ))
+
+soil_incr_vars=""
+for vi in $( seq 1 "${LSOIL_INCR}" ); do
+    soil_incr_vars=${soil_incr_vars}'"soilt'${vi}'_inc"',
+done
+for vi in $( seq 1 "${LSOIL_INCR}" ); do
+    soil_incr_vars=${soil_incr_vars}'"slc'${vi}'_inc"',
+done
+
+cat << EOF > regrid.nml
+ &config
+  n_vars=${n_vars},
+  variable_list=${soil_incr_vars}
+  missing_value=0.,
+ /
+ &input
+  gridtype="gau_inc",
+  ires=${LONB_CASE_IN},
+  jres=${LATB_CASE_IN},
+  fname="enkfgdas.sfci.nc",
+  dir="./",
+  fname_coord="gaussian_scrip.nc",
+  dir_coord="./"
+/
+
+ &output
+  gridtype="fv3_rst",
+  ires=${CRES_OUT},
+  jres=${CRES_OUT},
+  fname="sfci",
+  dir="./",
+  fname_mask="vegetation_type" 
+  dir_mask="./"
+  dir_coord="${FIXorog}",
+ /
+EOF
+
+# fixed input files
+ln -sf "${TMP_FIX_FILES}/gaussian.${LONB_CASE_IN}.${LATB_CASE_IN}.nc" "gaussian_scrip.nc"
+
+# fixed output files
+for n in $(seq 1 "${ntiles}"); do
+    ln -sf "${FIXorog}/${CASE_OUT}/sfc/${CASE_OUT}.mx${OCNRES_OUT}.vegetation_type.tile${n}.nc"  "vegetation_type.tile${n}.nc"
+done
+
+if (( LFHR >= 0 )); then 
+        soilinc_fhrs=("${LFHR}")
+else # construct restart times for deterministic member
+    soilinc_fhrs=("${assim_freq}") # increment file at middle of window 
+    if [[ "${DOIAU:-}" == "YES" ]]; then  # Update surface restarts at beginning of window
+      half_window=$(( assim_freq / 2 ))
+      soilinc_fhrs+=("${half_window}")
+    fi
+fi 
+
+for imem in $(seq 1 "${NMEM_REGRID}"); do
+    if (( NMEM_REGRID > 1 )); then
+        cmem=$(printf %03i "${imem}")
+        memchar="mem${cmem}"
+     
+        MEMDIR=${memchar} YMD=${PDY} HH=${cyc} declare_from_tmpl \
+            COMOUT_ATMOS_ANALYSIS_MEM:COM_ATMOS_ANALYSIS_TMPL
+
+        MEMDIR=${memchar} YMD=${PDY} HH=${cyc} declare_from_tmpl \
+            COMIN_SOIL_ANALYSIS_MEM:COM_ATMOS_ANALYSIS_TMPL
+    fi
+ 
+    for FHR in "${soilinc_fhrs[@]}"; do
+      ln -fs "${COMIN_SOIL_ANALYSIS_MEM}/${APREFIX_ENS}sfci00${FHR}.nc" \
+                "${DATA}/enkfgdas.sfci.nc"
+
+      "${APRUN_REGRID}" "${REGRID_EXEC}" "${REDOUT}${PGMOUT}" "${REDERR}${PGMERR}"
+
+      for n in $(seq 1 "${ntiles}"); do
+          mv "${DATA}/sfci.tile${n}.nc"  "${COMOUT_ATMOS_ANALYSIS_MEM}/sfci00${FHR}.tile${n}.nc" 
+      done
+    done
+done
```

Comment:
Another nit picky comment here. The indents are inconsistent.  Unfortunately there is not a linter that will catch this.

Reply:
done